### PR TITLE
adding visas controllers by Id

### PR DESCRIPTION
**Pull request by leoraba**

### CODE REVIEW

File: src/main/java/bio/overture/ego/controller/VisaController.java [hunk 1]
```diff
@@ -19,19 +19,21 @@
...
 import org.springframework.data.domain.Pageable;
 import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.*;
 
 @Slf4j
 @RestController
-@RequestMapping("/visa")
-@Tag(name = "Visa")
+@RequestMapping("/visas")
+@Tag(name = "Visas")
```
Comment (leoraba) [inline src/main/java/bio/overture/ego/controller/VisaController.java:36]: changing to plural controller mapping. to keep consistency with other controllers

File: src/main/java/bio/overture/ego/controller/VisaController.java [hunk 2]
```diff
@@ -71,6 +73,24 @@ public VisaController(
     }
   }
 
+  @AdminScoped
+  @RequestMapping(method = GET, value = "/{visaId}")
+  @ApiResponses(
+      value = {@ApiResponse(responseCode = "200", description = "Get Visa Using visa Id")})
+  @JsonView(Views.REST.class)
```
Comment (leoraba) [inline src/main/java/bio/overture/ego/controller/VisaController.java:80]: GET operation with visaID used by frontend

File: src/main/java/bio/overture/ego/controller/VisaController.java [hunk 3]
```diff
@@ -111,30 +131,28 @@ public VisaController(
    * @return Visa visa
    */
   @AdminScoped
-  @RequestMapping(method = PUT, value = "/{type}/{value}")
+  @RequestMapping(method = PUT, value = "/{visaId}")
   @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Update Visa")})
```
Comment (leoraba) [inline src/main/java/bio/overture/ego/controller/VisaController.java:135]: use the ID to update a single visa

File: src/main/java/bio/overture/ego/controller/VisaController.java [hunk 4]
```diff
@@ -111,30 +131,28 @@ public VisaController(
...
+    return visaService.update(visaId, visaRequest);
   }
 
   /*
    * This method is used to delete visa using visa id
    * @param visaId UUID
    */
   @AdminScoped
-  @RequestMapping(method = DELETE, value = "/{type}/{value}")
+  @RequestMapping(method = DELETE, value = "/{visaId}")
```
Comment (leoraba) [inline src/main/java/bio/overture/ego/controller/VisaController.java:149]: use the ID to delete a visa

**Review by Azher2Ali — APPROVED** [review]

**Review by joneubank — APPROVED** [review]
Looks great. Thanks for updating API naming for consistency.

**Pull request merged**